lib/messageUtils.ts

```diff
@@ -58,12 +60,20 @@ export function getDashboard(
   const encodedBranchSlug = branchSlug ? (encodeURIComponent(branchSlug)) : "";
   const dashboardId = getDashboardIdForTemplate(template);
 
+  // Showing the last 30 complete days to ensure the dashboard isn't including today which has no data yet
+  let encodedSubmissionDate = "30+day+ago+for+30+day";
+  if (startDate && endDate) {
+    encodedSubmissionDate = `${startDate}+to+${endDate}`;
```

Comment:
One thing to be careful of (and probably manually test to see if the docs are correct, and probably add a comment in the code for future changers of the code):

From https://cloud.google.com/looker/docs/filter-expressions#basic_structure_of_date_and_time_filters

> The initial time value is inclusive but the latter time value is not. So the expression 2018-05-18 12:00:00 to 2018-05-18 14:00:00 will return data with the time "2018-05-18 12:00:00" through "2018-05-18 13:59:59".

For example, I think the above expression would want to effectively be `endDate + 1`, though we'd have to convert to a date type to do the math.

Reply:
Yeah I noticed this in the dashbaords. I added a logic to +1 to the `proposedDuration` when calculated the `endDate`

Reply:
Please do encode all of the URL components, just to be safe (I think an earlier iteration of the patch did this).  The reasoning is that we don't closely track where this comes from, and (for example) the dates came from the network. Although we should be able to trust the network, we don't want to be vulnerable to injection attacks that were started upstream from us.